sycl/source/detail/platform_impl.cpp

```diff
@@ -19,37 +19,32 @@ namespace cl {
 namespace sycl {
 namespace detail {
 
-vector_class<platform>
-platform_impl_pi::get_platforms() {
-  vector_class<platform> platforms;
-
-  pi_uint32 num_platforms = 0;
-  PI_CALL(piPlatformsGet)(0, nullptr, &num_platforms);
-  info::device_type forced_type = detail::get_forced_type();
-
-  if (num_platforms) {
-    vector_class<RT::PiPlatform> pi_platforms(num_platforms);
-    PI_CALL(piPlatformsGet)(num_platforms, pi_platforms.data(), nullptr);
-
-    for (pi_uint32 i = 0; i < num_platforms; i++) {
-
-      platform plt =
-        detail::createSyclObjFromImpl<platform>(
-          std::make_shared<platform_impl_pi>(pi_platforms[i]));
-      // Skip platforms which do not contain requested device types
-      if (!plt.get_devices(forced_type).empty())
-        platforms.push_back(plt);
-    }
+vector_class<platform> platform_impl::get_platforms() {
+  vector_class<platform> Platforms;
+
+  pi_uint32 NumPlatforms = 0;
+  PI_CALL(piPlatformsGet)(0, nullptr, &NumPlatforms);
+  info::device_type ForcedType = detail::get_forced_type();
+
+  if (NumPlatforms) {
+    vector_class<RT::PiPlatform> PiPlatforms(NumPlatforms);
+    PI_CALL(piPlatformsGet)(NumPlatforms, PiPlatforms.data(), nullptr);
+
+    std::for_each(PiPlatforms.cbegin(), PiPlatforms.cend(),
+                  [&](const RT::PiPlatform &PiPlatform) {
+                    platform Platform = detail::createSyclObjFromImpl<platform>(
+                        std::make_shared<platform_impl>(PiPlatform));
+                    // Skip platforms which do not contain requested device
+                    // types
+                    if (!Platform.get_devices(ForcedType).empty())
+                      Platforms.push_back(Platform);
+                  });
   }
-  return platforms;
-}
 
-vector_class<device>
-platform_impl_host::get_devices(info::device_type dev_type) const {
-  vector_class<device> res;
-  if (dev_type == info::device_type::host || dev_type == info::device_type::all)
-    res.resize(1); // default device construct creates host device
-  return res;
+  if (ForcedType == info::device_type::host || ForcedType == info::device_type::all)
+    Platforms.emplace_back(platform());
```

Comment:
According to the description in header file, this function must always return host platform.

Suggest adjusting comment that we have some mechanism to override this behavior